### DISCUSSION TITLE
Unable to compile on macOS Big Sur  "implicit declaration of function 'hmac_memory' is invalid in C99"

### DISCUSSION THREAD

**Discussion by Casuallynoted** [stepmania/stepmania#2110] [Q&A]
Hello!

I'm getting the following issue when compiling on macOS Big Sur:
"implicit declaration of function 'hmac_memory' is invalid in C99"

A warning that happens earlier in the build process:
"warning: 
      'LTC_YARROW_AES' macro redefined [-Wmacro-redefined]"

**Comment by concubidated** [comment]
OSX Support is pretty crippled on the mainline repo right now. Outfox supports the latest OSX, I would recommend checking it out.

https://projectmoon.dance/

**Comment by coderjo** [comment]
It sounds like part of the needed libtomcrypt files are not in your project file.

**Comment by koganikor** [comment]
Got it compiled on 11.2.3 installing many libraries with brew and with the options "WITH_SYSTEM_***"
